src/libvncserver/main.c

```diff
@@ -609,6 +609,11 @@ clientInput(void *data)
     cl->sock = RFB_INVALID_SOCKET;
 
     rfbClientConnectionGone(cl);
+	
+#ifdef LIBVNCSERVER_HAVE_LIBPTHREAD
+    // We can't join this thread, detach it here so the associated memory will be feed
+    pthread_detach(cl->client_thread);
```

Comment:
Two thoughts here:

* Maybe rather `pthread_detach(pthread_self());` for more clarity?
* What about `LIBVNCSERVER_HAVE_WIN32THREADS`? That might need a fix as well, the original code only did a `_beginthread()` and that was it. Only found https://stackoverflow.com/questions/12744324/how-to-detach-a-thread-on-windows-c that mentions something...

Reply:
1. I am unsure this is more clear. But I think it will work. However it would be best to test first (as the other code was tested). For me it is at the moment not so easy to test,
2. Sadly I have no knowledge of windows threads. A similar fix might be needed. The pull request however only fixes the pthread memory leak.